internal/dataplane/parser/ingressrules.go

```diff
@@ -299,33 +300,33 @@ func servicesAllUseTheSameKongAnnotations(
 		// validation can work. We should be able to move this validation there
 		// once https://github.com/Kong/kubernetes-ingress-controller/issues/2195
 		// is resolved.
+
 		for k, v := range annotations {
 			valueForThisObject, ok := service.Annotations[k]
 			if !ok {
-				failuresCollector.PushResourceFailure(
-					fmt.Sprintf("Service has inconsistent %s annotation and is used in multi-Service backend %s. "+
-						"All Services in a multi-Service backend must have matching Kong annotations. Review the "+
-						"associated route resource and align annotations in its multi-Service backends.",
-						k, kongServiceName),
-					service.DeepCopy(),
-				)
+				badAnnotations[k] = nil
 				match = false
 				// continue as it doesn't make sense to verify value of not existing annotation
 				continue
 			}
 
 			if valueForThisObject != v {
-				failuresCollector.PushResourceFailure(
-					fmt.Sprintf("Service has inconsistent %s annotation and is used in multi-Service backend %s. "+
-						"All Services in a multi-Service backend must have matching Kong annotations. Review the "+
-						"associated route resource and align annotations in its multi-Service backends.",
-						k, kongServiceName),
-					service.DeepCopy(),
-				)
+				badAnnotations[k] = nil
 				match = false
 			}
 		}
 	}
 
+	for _, service := range services {
+		for annotation := range badAnnotations {
+			failuresCollector.PushResourceFailure(
+				fmt.Sprintf("Service has inconsistent %s annotation and is used in multi-Service backend %s. "+
+					"All Services in a multi-Service backend must have matching Kong annotations.",
```

Comment:
It's kinda buried, but we only include ours when generating the list earlier, in getK8sServicesForBackends: https://github.com/Kong/kubernetes-ingress-controller/blob/696729bcbe8d87b926d7a03be710d435cbfdd7ea/internal/dataplane/parser/ingressrules.go#L273

Added some comments earlier in the call stack to note when the filtering happens, changed `servicesAllUseTheSameKongAnnotations` to `collectInconsistentAnnotations`, and added a comment to indicate that it's just using the annotation list given rather than implying that it's handling filtering.